modules/dmrpp_module/DMZ.cc

```diff
@@ -1517,6 +1518,229 @@ DMZ::process_missing_data(BaseType *btp, const xml_node &missing_data)
  
 }
 
+bool 
+DMZ::supported_special_structure_type_internal(Constructor *var_ctor) {
+
+    bool ret_value = true;
+    Constructor::Vars_iter vi = var_ctor->var_begin();
+    Constructor::Vars_iter ve = var_ctor->var_end();
+    for (; vi != ve; vi++) {
```

Comment:
If you use the `const vector<BaseType*> &variables() const { return d_vars; }` method, these loops can be re-written using the range-based for loop:

for (auto bt: var_ctor->variables()) {
...
}